sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/InvalidCastToInterface.cs

```diff
@@ -132,4 +131,44 @@ public void TestMethod(object obj)
             return null;
         }
     }
+
+    interface IFoo { }
+    interface IBar { }
+
+    class Foo : IFoo { }
+    class Bar : IBar { }
+    class FooBar : IFoo, IBar { }
+    sealed class FinalBar : IBar { }
+
+    class Other
+    {
+        public void Method()

```

Comment:
I just figured out that we did not talk about generic arguments (with and without constraints) - AFAIK the compiler already raises errors when a variable cannot be cast to generic argument, perhaps we should extend our checks to raise code smell for that as well? Do you think this is reasonable?